tests/FakeItEasy.Tests/Core/DefaultSutInitializerTests.cs

```diff
@@ -58,11 +59,10 @@ public void Should_use_same_instance_when_multiple_arguments_are_of_same_type()
         public void Should_pass_empty_options_builder_to_fake_manager_for_each_constructor_argument()
         {
             // Arrange
-            this.StubFakeManagerWithFake<IFoo>();
-            this.StubFakeManagerWithFake<object>();
             var optionsBuilders = new List<Action<IFakeOptions>>();
             A.CallTo(() => this.fakeManager.CreateFake(A<Type>._, A<Action<IFakeOptions>>._))
-                .Invokes((Type type, Action<IFakeOptions> optionBuilder) => optionsBuilders.Add(optionBuilder));
+                .Invokes((Type type, Action<IFakeOptions> optionBuilder) => optionsBuilders.Add(optionBuilder))
+                .ReturnsLazily((Type type, Action<IFakeOptions> fakeOptionBuilder) => Create.Fake(type));

```

Comment:
Without the `ReturnsLazily`, our new `Invokes` behaviour returns a fake `object`, which is incompatible with the expected return type.
We're just simulating the "stubbing" behaviour we had before.